### PR TITLE
Switch to 3.8 as the base Python version

### PR DESCRIPTION
**Pull request by ghickman**
This sets the minimum version of Python we support to 3.8.

### CODE REVIEW

File: jobrunner/reusable_actions.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def handle_reusable_action(run_command):
 
     reusable_action = fetch_reusable_action(image, tag)
     new_run_args = apply_reusable_action(run_args, reusable_action)
-    # When we no longer support Python 3.7 we can just use `shlex.join()` here
+    # we can't use `shlex.join()` here because it fails on Windows
```
Comment (iaindillingham) [inline jobrunner/reusable_actions.py:87]: That's a pain.

**Review by evansd — APPROVED** [review]
Looks good. Thanks for doing this! I think if you grep for `3.8` you'll find some more things that can be dropped/simplified, but we don't have to do these now.

**Review by iaindillingham — APPROVED** [review]

File: jobrunner/reusable_actions.py (hunk 1 above)
Comment (evansd) [reply jobrunner/reusable_actions.py:87]: The trackback makes it look like that failure was under 3.7:
```
C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\shutil.py:400: in _rmtree_unsafe
```

Are you sure `shlex.join()` was the culprit here?

Comment (ghickman) [reply jobrunner/reusable_actions.py:87]: Well now I'm not 😰

Comment (ghickman) [reply jobrunner/reusable_actions.py]: Yep, looks like I just messed it up, working in the latest actions run.

**Pull request merged**